src/cloudai/workloads/ai_dynamo/slurm_command_gen_strategy.py

```diff
@@ -52,18 +75,45 @@ def _get_toml_args(self, base_model: BaseModel, prefix: str, exclude: List[str]
     def _gen_script_args(self, td: AIDynamoTestDefinition) -> List[str]:
         args = []
 
+        huggingface_path = (
+            td.cmd_args.huggingface_home_container_path
+            if td.cmd_args.dynamo.backend == "vllm"
+            else td.cmd_args.huggingface_home_host_path
+        )
         args.extend(
             [
-                f"--huggingface-home {td.cmd_args.huggingface_home_container_path}",
+                f"--huggingface-home {huggingface_path}",
                 "--results-dir /cloudai_run_results",
             ]
         )
 
+        # Get base dynamo args first
         args.extend(
             self._get_toml_args(
                 td.cmd_args.dynamo, "--dynamo-", exclude=["prefill_worker", "decode_worker", "genai_perf"]
             )
         )
+
+        # Add backend-specific args
+        if td.cmd_args.dynamo.backend == "sglang":
+            dynamo_repo_path = td.dynamo_repo.installed_path
+            if dynamo_repo_path is None:
+                raise ValueError("dynamo_repo_path is not set - repo may not be installed")
+
+            deepep_path = (
+                dynamo_repo_path / "components/backends/sglang/configs/deepseek_r1/wideep/deepep.json"
+            ).absolute()
+            sgl_http_server_path = (
+                dynamo_repo_path / "components/backends/sglang/src/dynamo/sglang/utils/sgl_http_server.py"
+            ).absolute()
+
+            args.extend(
+                [
+                    f'--dynamo-sgl-http-server-script "{sgl_http_server_path!s}"',
```

Comment:
This all becomes redundant once we move to using extra container mounts and parameters to specify filepaths.